src/algebra/module/linear_map.lean

```diff
@@ -133,6 +133,9 @@ variables (f g)
 @[simp] lemma map_zero : f 0 = 0 :=
 f.to_distrib_mul_action_hom.map_zero
 
+lemma zero_of_injective_zero (h : function.injective f) {x : M} (w : f x = 0) : x = 0 :=
```

Comment:
Could you turn this into `map_eq_zero_iff : f x = 0 ↔ x = 0`?

Not directly related to the PR, but should be addressed at some point: We a result like this for various `hom`s stated in various ways with various names: `add_monoid_hom` and `ring_hom` have `injective_iff.mp`, `add_equiv` and `linear_equiv` have `map_eq_zero_iff`, `pseudoelements` has `zero_of_map_zero`, and there's probably a few that I'm missing.

Reply:
Done!